lib/statsd/instrument/assertions.rb

```diff
@@ -49,19 +49,34 @@ module Assertions
       # @param [Array<String>] metric_names (default: []) The metric names that are not
       #   allowed to happen inside the block. If this is set to `[]`, the assertion
       #   will fail if any metric occurs.
+      # @param [Array<StatsD::Instrument::Datagram>] datagrams (default: nil) The datagrams
+      #   to be inspected for metric emission.
+      # @param [StatsD::Instrument::Client] client (default: nil) The client to be used
+      #   for fetching datagrams (if not provided) and metric prefix. If not provided, the
+      #   singleton client will attempt to be used.
+      # @param [Boolean] no_prefix (default: false) A directive to indicate if the client's
```

Comment:
I think `prefix: false` would be much more idomatic.

Reply:
I do think the positive case would read better, but all the existing assertions that take parameters for this take it in the form of `no_prefix` and I'd rather keep consistency.